sql/core/src/main/scala/org/apache/spark/sql/execution/merge/MergeScalarSubqueries.scala

```diff
@@ -0,0 +1,627 @@
+/*
```

Comment:
I moved this file from `catalyst` to `sql` in the first commit: https://github.com/apache/spark/pull/37630/commits/f53cddd148e5c6584570393084f07a4e036e342e, but unfortunately Git doesn't recognize the move as there are too many additions to the file: https://github.com/apache/spark/pull/37630/commits/269c75be30d262958c125c063482674cc0d99e00